DomainContent/Rust/DJ-Tools/DJ_Input_Zone_Client.js

```diff
@@ -0,0 +1,382 @@
+        var entList = Entities.findEntities(currentProps.position,searchRadius);
+        entList.forEach(function(ent) {
+            var userData = gProps(ent, ["userData"]).userData;
+            try {
+                userData = JSON.parse(userData);
+                if (userData.performance) {
+                    if (userData.performance.type === DEBUG_CUBE) {
+                        var inputProps = gProps(ent);
+                        var name = inputProps.name;
+                        var id = inputProps.id;
+                        matchedEnts.push(
+                            new Debug(
+                                name, 
+                                id
+                            )
+                        );
+                    }
+                }
+            } catch (e) {
+                console.log(e);
+            }
+        });
+        return matchedEnts;
+    }
+
+    function updateInputsEndPoints() {
+        inputs.forEach(function(input) {
+            input.getEndPoints();
+        });
+    }
+
+    function updateInputsAndEndPoints() {
+        inputs = scanForInputs();
+        endPoints = scanForEndpoints();
+        debugs = scanForDebugs();
+        updateInputsEndPoints();
+    }
+
+    function onUpdate(delta) {
+        deltaTotal += delta;
+        if (deltaTotal > DELTA_UPDATE_INTERVAL) {
+            inputs.forEach(function (input) {
+                [LEFT_HAND, RIGHT_HAND, DEBUG_CUBE].forEach(function (target) {
+                    targetPosition = getTargetPosition(target);
+                    result = checkIfIn(targetPosition, input);
+                    check = returnCheck(target);
+                    if (target === DEBUG_CUBE) {
+                        debugInfo.result = result;
+                        debugInfo.check = check;
+                    }
+                    if (result) {
+                        if (!input[check]) {
+                            input[check] = true;
+                            input.sendOn();
+                            
+                        } else {
+                            input.sendEdit(targetPosition);
+                        }
+                    } else {
+                        if (input[check]) {
+                            input.sendOff();
+                            input[check] = false;
+                        }
+                    }
+                });
+            });
+        }
+
+        deltaTotal = 0;
+    }
+
+    // Entity Definition
+    function DJ_Input_Zone_Client() {
+    }
+
+    DJ_Input_Zone_Client.prototype = {
+        preload: function (id) {
+            _entityId = id;
+            currentProps = Entities.getEntityProperties(id);
+            userData = currentProps.userData;
+            userdataProperties = JSON.parse(userData);
+            updateInputsAndEndPoints();
+        },
+        enterEntity: function () {
+            console.log("ENTERING ZONE");
```

Comment:
Wrap in debug statement or remove prints

Reply:
done